lib/posix-process: Bugfixes and warnings

michpappas opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

* Add missing initialization of cl_args in uk_posix_clone_sighand()
* Fix clone return value overwrite
* Fix warnings triggered on different configs

michpappas committed:
lib/posix-process/signal: Initialize cl_args before use

Add missing initialization of cl_args in uk_posix_clone_sighand()

Signed-off-by: Michalis Pappas <michalis@unikraft.io>

mogasergiu reviewed (approved):
Reviewed-by: Sergiu Moga <sergiu@unikraft.io>
Approved-by: Sergiu Moga <sergiu@unikraft.io>

michpappas committed:
lib/posix-process: Fix warnings triggered on different configs

Fix warnings triggred when execve or signals are not enabled.

Signed-off-by: Michalis Pappas <michalis@unikraft.io>

michpappas committed:
lib/posix-process: Fix clone return value overwrite

Move assingment of return value after uk_sched_thread_add()
to avoid overwriting the return value.

Signed-off-by: Michalis Pappas <michalis@unikraft.io>

mogasergiu reviewed (approved):
Reviewed-by: Sergiu Moga <sergiu@unikraft.io>
Approved-by: Sergiu Moga <sergiu@unikraft.io>
